File: api/src/main/java/com/aa/weatherprediction/model/City.java
package com.aa.weatherprediction.model;

import com.squareup.moshi.JsonClass;
import kotlinx.serialization.Serializable;

@JsonClass(generateAdapter = true)
@Serializable
public record City(
        String name,
        Double lat,
        Double lon,
        String country,
        String state
) {
    public City() {
        this(null, null, null, null, null);
    }
}

File: weather-prediction-api/src/main/kotlin/com/aa/weatherprediction/model/City.kt
package com.aa.weatherprediction.model

import com.squareup.moshi.JsonClass
import kotlinx.serialization.Serializable

@JsonClass(generateAdapter = true)
@Serializable
data class City(
    var name: String?,
    var lat: Double?,
    var lon: Double?,
    var country: String?,
    var state: String?,
) {
    constructor() : this(null, null, null, null, null)
}
